src/component/Product.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import "./Product.css";
+import { useStateValue } from "../context/StateProvider";
+
+import Pagination from "./Pagination";
+import { Link } from "react-router-dom";
+
+function Product({ products }) {
+  const [{ basket, user, search }, dispatch] = useStateValue();
+  const [items, setItem] = useState(products);
+
+  const [loging, setLoding] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [postsPerPage, setPostPerPage] = useState(6);
+
+  let filterdProducts = products;
+
+  if (search != "") {
+    filterdProducts = products?.filter((i) => {
+      return (
+        i.title.toLowerCase().indexOf(search.toLowerCase()) !== -1 ||
+        i.description.toLowerCase().indexOf(search.toLowerCase()) !== -1 ||
+        i.category.toLowerCase().indexOf(search.toLowerCase()) !== -1
+      );
+    });
+  }
+
+  const IndexOfLastPost = currentPage * postsPerPage;
+  const indexOfFirstPost = IndexOfLastPost - postsPerPage;
+  const currentPosts = filterdProducts.slice(indexOfFirstPost, IndexOfLastPost);
+
+  const paginate = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
+
+  return (
+    <div>
+      <div className="product">
+        {currentPosts.map((product, index) => {
+          return (
+            <Link
+              to={{
+                pathname: "/productdetail/"+product.id,
```

Comment:
you could use template strings instead of the older JS way of concatenating strings.